File: 268. Missing Number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    var res = 0;
    for (var i = 1; i <= nums.length; i++)
        res ^= i;
    for (i = 0; i < nums.length; i++)
        res ^= nums[i];
    return res;
};

File: 268. Missing Number.py
class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        res = 0
        for i in xrange(1, n+1):
            res ^= i
        for i in nums:
            res ^= i
        return res
